jsonevtloop_jsonschema_V2.txt

```diff
@@ -0,0 +1,153 @@
+
+	-----------------------------------------
+	Description of the jsonevtloop and schema
+	-----------------------------------------
+
+Purpose: JSON-based asynchronous API for signal-cli.
+
+This mode is not intended to be used directly from the command line, but to 
+provide an easy use API for other scripting languages.
+
+signal-cli listens for JSON requests on stdin. Each JSON request must be
+a single valid JSON object (enclosed in {}), followed by LF. The JSON text
+itself hence must not contain linefeeds (only escaped as \n and no pretty
+printing enabled).
+
+When incoming messages are received from the signal back-end, they are
+in turned JSON encoded and sent out on stdout.
+
+Any error messages or debug information is sent out on stderr as expected.
+
+
+
+JSON Schema Version 2 
+
+
+	JSON Requests are read from STDIN in "jsonevtloop" mode and the Reponses are
+	sent to STDOUT in "jsonevtloop" mode
+	
+	Generic Request attributes valid for ALL requests
+	{
+		"reqType": "<request-type>",				// MANDATORY: Name of request type
```

Comment:
Not particularly important, but say you debug stuff, or develop, you always end up with a bunch of JSON dumps on screen or in textfiles etc and this way it's really clear what are requests and what are responses, a context otherwise lost...